plotly/figure_factory/_dendrogram.py

```diff
@@ -140,7 +140,10 @@ def __init__(self, X, orientation='bottom', labels=None, colorscale=None,
         for i in range(len(yvals_flat)):
             if yvals_flat[i] == 0.0 and xvals_flat[i] not in self.zero_vals:
                 self.zero_vals.append(xvals_flat[i])
-
+        if len(self.zero_vals) > len(yvals) + 1:
+            l_border = int(min(self.zero_vals))
+            r_border = int(max(self.zero_vals))
+            self.zero_vals = [v for v in range(l_border,r_border + 1, int((r_border-l_border) / len(yvals)))]
```

Comment:
Please break this line up into two lines so that it's not so wide. Maybe create a new local variable for the `range(...)` expression.